lib/Dialect/Pulse/Transforms/Scheduling.cpp

```diff
@@ -125,27 +125,28 @@ void QuantumCircuitPulseSchedulingPass::scheduleAlap(
                               << quantumGateCallSequenceOpDuration << "\n");
 
       // find next available time for all the ports
-      const int nextAvailableTimeOfAllPorts =
+      const int64_t nextAvailableTimeOfAllPorts =
           getNextAvailableTimeOfPorts(ports);
       LLVM_DEBUG(llvm::dbgs() << "\t\tnext availability is at "
                               << nextAvailableTimeOfAllPorts << "\n");
 
       // find the updated available time, i.e., when the current quantum gate
       // will be scheduled
-      const int updatedAvailableTime =
+      int64_t updatedAvailableTime =
           nextAvailableTimeOfAllPorts - quantumGateCallSequenceOpDuration;
+      // set the timepoint of quantum gate
+      PulseOpSchedulingInterface::setTimepoint(quantumGateCallSequenceOp,
+                                               updatedAvailableTime);
       LLVM_DEBUG(llvm::dbgs() << "\t\tcurrent gate scheduled at "
                               << updatedAvailableTime << "\n");
       // update the port availability map
+      if (quantumGateCallSequenceOp.getCallee().str().find("measure") == 0)
```

Comment:
Is this relying on the sequence name containing the word `measure` rather than detecting the presence of a `pulse.capture`? If someone were to use a sequence that did not have `measure` in the name would this fail?